lib/Model/Listing/Dao/QueryBuilderHelperTrait.php

```diff
@@ -120,45 +112,34 @@ private function applyOrderByToQueryBuilder(QueryBuilder $queryBuilder): void
         }
     }
 
-    /**
-     * @param DoctrineQueryBuilder $queryBuilder
-     */
     private function applyLimitToQueryBuilder(QueryBuilder $queryBuilder): void
     {
         $queryBuilder->setFirstResult($this->model->getOffset());
         $queryBuilder->setMaxResults($this->model->getLimit());
     }
 
-    protected function prepareQueryBuilderForTotalCount(DoctrineQueryBuilder &$queryBuilder): void
+    protected function prepareQueryBuilderForTotalCount(QueryBuilder $queryBuilder, string $identifierColumn): void
```

Comment:
Note: this will create a backwards incompatibility when this trait is used within application landscape. If you prefer, I could rewrite this to a runtime error instead

```
protected function prepareQueryBuilderForTotalCount(QueryBuilder $queryBuilder, ?string $identifierColumn = null): void
{
    // ...
    } elseif ($this->isQueryBuilderPartInUse($queryBuilder, 'distinct')) {
        if ($identifierColumn === null) {
            throw new \InvalidArgumentException(...);
        }
        // ...
    }
}
```

Reply:
`QueryBuilderHelperTrait` is `@internal`, so should be fine ...

Reply:
I suddenly notice I made a fuck up here anyway (`$queryBuilder` was passed as reference with a good reason), will have a look tomorrow with some fresh energy

Reply:
Updated my PR, added unit test as well so we have this part covered in the future